File: app/src/main/java/com/rdev/tryp/blocks/forme/edit_addresses/AddressEditor.java
package com.rdev.tryp.blocks.forme.edit_addresses;

import android.app.Activity;
import com.rdev.tryp.R;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

public class AddressEditor implements Editor.IEditor {

    @Override
    public void editAddresses(Activity activity, Editor.IView view) {
        Fragment fragment = new EditAddressesFragment(view);
        ((AppCompatActivity)activity).getSupportFragmentManager().beginTransaction()
                .add(R.id.screenContainer, fragment)
                .addToBackStack("edit_address")
                .commit();
    }

}

File: app/src/main/java/com/rdev/tryp/blocks/forme/edit_addresses/AddressEditor.kt
package com.rdev.tryp.blocks.forme.edit_addresses

import android.app.Activity
import com.rdev.tryp.R

import androidx.appcompat.app.AppCompatActivity


class AddressEditor : Editor.IEditor {

    override fun editAddresses(activity: Activity, view: Editor.IView) {
        val fragment = EditAddressesFragment(view)
        (activity as AppCompatActivity).supportFragmentManager.beginTransaction()
                .add(R.id.screenContainer, fragment)
                .addToBackStack("edit_address")
                .commit()
    }

}